internals/overlord/checkstate/manager_test.go

```diff
@@ -869,3 +870,133 @@ func (s *ManagerSuite) TestReplan(c *C) {
 	c.Assert(status, Matches, "Do.*")
 	c.Assert(change.Kind(), Equals, "perform-check")
 }
+
+func (s *ManagerSuite) TestMetricsCheckSuccess(c *C) {
+	tempDir := c.MkDir()
+	tempFile := filepath.Join(tempDir, "file.txt")
+	command := fmt.Sprintf(`/bin/sh -c 'echo -n x >>%s'`, tempFile)
+	s.manager.PlanChanged(&plan.Plan{
+		Checks: map[string]*plan.Check{
+			"chk1": {
+				Name:      "chk1",
+				Override:  "replace",
+				Period:    plan.OptionalDuration{Value: 10 * time.Millisecond},
+				Timeout:   plan.OptionalDuration{Value: time.Second},
+				Threshold: 3,
+				Exec:      &plan.ExecCheck{Command: command},
+				Startup:   plan.CheckStartupEnabled,
+			},
+		},
+	})
+
+	// Wait for check to run at least twice
+	for i := 0; ; i++ {
+		if i >= 1000 {
+			c.Fatalf("failed waiting for check to run")
+		}
+		b, _ := os.ReadFile(tempFile)
+		if len(b) >= 2 {
+			break
+		}
+		time.Sleep(time.Millisecond)
+	}
+
+	buf := new(bytes.Buffer)
+	writer := metrics.NewOpenTelemetryWriter(buf)
+	s.manager.WriteMetrics(writer)
+	expected := `
+# HELP pebble_check_up Whether the health check is up (1) or not (0)
+# TYPE pebble_check_up gauge
+pebble_check_up{check="chk1"} 1
+
+# HELP pebble_check_success_count Number of times the check has succeeded
+# TYPE pebble_check_success_count counter
+pebble_check_success_count{check="chk1"} 2
```

Comment:
Hmm, this is racy. If CI is slow and the check (which has a short period) runs 3 or 4 times, this will fail. The simplest way I can think of is using a regex to match , with `\d+` in this position. (I think any number, even 1 here, would be okay. I don't think there's a specific reason we need to wait for it to run twice.)